test/IntegrationTests/FileVectorizer/test_deletefiles.py

```diff
@@ -0,0 +1,363 @@
+                response = requests.delete(
+                    delete_url,
+                    params={"filename": filename},
+                    headers=customer["headers"]
+                )
+                self.assertEqual(response.status_code, 200, f"❌ Failed to delete {filename}")
+                logger.info(f"Initiated deletion for {filename}")
+
+            # Then monitor status
+            last_control_time = time.time()
+            control_interval = 30
+            start_time = time.time()
+            deleted_files = set()
+
+            while True:
+                try:
+                    current_time = time.time()
+
+                    # Vectorizer control (if needed)
+                    if current_time - last_control_time >= control_interval:
+                        logger.info("Stopping file vectorizer...")
+                        self.control_file_vectorizer()
+                        last_control_time = current_time
+
+                    # Check deletion status
+                    response = requests.get(
+                        deletion_status_url,
+                        headers=customer["headers"],
+                        params={"page": 1, "page_size": 100}
+                    )
+                    self.assertEqual(response.status_code, 200)
+
+                    # Process all files in response
+                    for file in response.json():
+                        filename = file.get("filename")
+                        if filename in customer_filenames:
+                            status = file.get("deletion_status")
+                            if status == "DELETION_COMPLETED":
+                                deleted_files.add(filename)
+                                logger.info(f"Deletion completed for {filename}")
+                            else:
+                                logger.info(f"{filename} status: {status}")
+
+                    # Check completion
+                    remaining = customer_filenames - deleted_files
+                    if not remaining:
+                        logger.info(f"Customer {idx + 1}: All files deleted in {time.time() - start_time:.1f}s!")
+                        break
+                    else:
+                        logger.info(f"Customer {idx + 1} - Waiting for {len(remaining)} files...")
+
+                    time.sleep(2)
+
+                except Exception as e:
+                    logger.error(f"Deletion monitoring error: {str(e)}")
+                    time.sleep(5)
+
+    def control_file_vectorizer(self):
+        container_name = "chat_service"
+        try:
+            subprocess.run( ["docker", "exec", container_name, "pkill", "-9", "-f","file_vectorize_main"],check=True)
+            logger.info("Successfully killed file_vectorize_main process with pkill -9")
+        except subprocess.CalledProcessError as e:
+            logger.error(f"Failed to kill file_vectorize_main: {e}")
+
```

Comment:
Add another testcase where you add 10 and delete those 10 files at random times. There should be no correlation like that you are first adding 10 files and then deleting 10 files. The flow should be completely random like below 

Add a code to bring this randomness for the 10 files like below. The time slot between every add or delete can be 10 seconds
Like add file1
add file 2
delete file1
add file 3
delete file3
delete file2